### ISSUE TITLE
Example does not work with go modules

### ISSUE THREAD

**Issue by ritesh** [awslabs/goformation#206]
Hi, 

I'm trying to vendor `goformation` as a dependency. To do this, i'm using `go mod`. When I try to `go build` the example in the README, it fails. 

Steps to reproduce:
- Create a new directory (e.g. /tmp/hello) outside of `$GOPATH`
- Initialise a new go module `go mod init github.com/ritesh/mycoolmodule`
- Create a `main.go` file using the example provided in the README of this repo
- Run `go build`. 

Error message
```
build github.com/ritesh/mycoolmodule: cannot load github.com/awslabs/goformation/cloudformation/resources: cannot find module providing package github.com/awslabs/goformation/cloudformation/resources
```

Expected result. I would expect `go mod` to automatically fetch `goformation` and the `resources` module under it and build my code. I suspect this is `go mod` deficiency and I'm happy to explore other ways to vendor `goformation`

**Comment by stilvoid** [comment]
After some chatting about this, we found out that it's because (for whatever reason) `go mod` is picking up `v1.4.1` as the latest revision of `goformation` (GitHub shows the latest tag as being `v2.3.0`).

**Comment by flemay** [comment]
@ritesh I got it working with the following in `go.mod`: `require github.com/awslabs/goformation v2.3.0`.

However, when building/running it, `v.2.3.0` will be changed to `v0.0.0-20190320125420-ac0a17860cf1`, but the example works!

@stilvoid `Go modules` don't automatically follow github releases. `Version 2` has breaking changes. Doing `go get github.com/awslabs/goformation` will download latest of `v1` and not break existing software that requires it.

Some changes will need to be made in this repo to reflect `Version 2`. This [post](https://medium.com/@aman.sardana/go-modules-versioning-dependency-management-d5f96b490774) explains it quite well.

I don't think it is simply *for whatever reason* and currently people who want to use `goformation` can't make the example works out of the box.

**Comment by ritesh** [comment]
Thanks for that @flemay. Can't say i'm a fan of `go mod` so far, as i still don't fully grok these weird dependency resolution cases (maybe i just don't get semver).

**Comment by flemay** [comment]
@ritesh `semver` is quite simple to understand on paper. However the implementation of it in Go is trickier at first. You don't need a central registry (like NPM) to download your dependencies and you can use different versions of a dependency in your code.

So ideally if you want to use `v2` of goformation, you would import `"github.com/awslabs/goformation/cloudformation/v2"`. ( *this does not work at the moment*)

This [medium post](https://medium.com/rungo/anatomy-of-modules-in-go-c8274d215c16) explains in details semver/go modules.

**Comment by alock** [comment]
Any plan to update this module to be `v2` module since the semver is there?  With a 1.13 project I cannot use this repo as a dependency error below.

```shell
$ go mod download
go: finding github.com/awslabs/goformation v2.3.0+incompatible
go: finding github.com/awslabs/goformation v2.3.0+incompatible
github.com/awslabs/goformation@v2.3.0+incompatible: invalid version: +incompatible suffix not allowed: module contains a go.mod file, so semantic import versioning is required
```

**Comment by PaulMaddox** [comment]
Marking this as resolved. This was fixed in v2.3.1 and since then v3.0.0 has also been released. 
This was totally down to my ignorance of go module versioning when you hit v2, and shouldn't happen again :)
